backend/dataall/modules/datasets_base/db/dataset_models.py

```diff
@@ -141,3 +141,23 @@ class Dataset(Resource, Base):
     @classmethod
     def uri(cls):
         return cls.datasetUri
+
+
+class DatasetBucket(Resource, Base):
+    __tablename__ = 'dataset_bucket'
+    datasetUri = Column(String, nullable=False)
+    bucketUri = Column(String, primary_key=True, default=utils.uuid('bucket'))
+    AwsAccountId = Column(String, nullable=False)
+    S3BucketName = Column(String, nullable=False)
+    region = Column(String, default='eu-west-1')
+    partition = Column(String, default='aws')
+    KmsAlias = Column(String, nullable=False)
+    imported = Column(Boolean, default=False)
+    importedKmsKey = Column(Boolean, default=False)
+    userRoleForStorageBucket = query_expression()
+    projectPermission = query_expression()
+    environmentEndPoint = query_expression()
```

Comment:
Are the following fields used in `DatasetBucket` table: `partition` , `userRoleForStorageBucket`, `projectPermission`, and `environmentEndPoint` ?

Reply:
I believe we had a discussion on this in weekly calls. It was decided that we would keep this class uniform with other classes in this file and it would be addressed by a database/table clean up that AWS would be undertaking separately.